File: src/main/python/easy/RansomNote.py
from collections import Counter


class RansomNote:
    def canConstruct(self, ransomNote, magazine):
        st1, st2 = Counter(ransomNote), Counter(magazine)
        if st1 & st2 == st1:
            return True
        return False


test = RansomNote()
print(test.canConstruct("aa", "aba"))

File: src/main/java/easy/RansomNote.java
package easy;

public class RansomNote {
    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.canConstruct("aa","aba"));
    }

    static class Solution {
        public boolean canConstruct(String ransomNote, String magazine) {
            if (ransomNote.isEmpty() || magazine.isEmpty() || ransomNote.length() > magazine.length())
                return false;
            int[] counter = new int[26];

            for (char c : magazine.toCharArray())
                counter[c - 'a']++;

            for (char c : ransomNote.toCharArray()) {
                if (counter[c - 'a'] == 0)
                    return false;
                counter[c - 'a']--;
            }
            return true;
        }
    }
}

